src/FormatNumber/Parser.elm

```diff
@@ -378,3 +380,40 @@ parse locale original =
                 | prefix = locale.zeroPrefix
                 , suffix = locale.zeroSuffix
             }
+
+{-| Given a `Locale` parses a `String` into a `Maybe Float`:
+-}
+parseString : Locale -> String -> Maybe Float
+parseString locale value = 
+    let
+        isNegative : Bool
+        isNegative = 
+            if String.left 1 value == locale.negativePrefix then
+                True
+            else
+                False
+        chrs : Regex.Regex
+        chrs = 
+            ["[", locale.negativePrefix, locale.negativeSuffix, locale.thousandSeparator, "]"]
+                |> String.concat
+                |> Regex.fromString
+                |> Maybe.withDefault Regex.never
```

Comment:
I think this strategy might be overcomplicated and potentially wrong since `negativePrefix`, `negativeSuffix` and `thousandSeparatot` might
1. compromise the regular express syntax and
2.  be misleading if longer than 1 character.

What I would try is:
- Split the number by `decimalSeparator`, i.e. `"-3,141.5" => ("-3,141", "5")`
- Clean up the parts, i.e. `("-3,141", "5") => ("3141", "5")`
- Parse the parts to `Int`, i.e. `("3141", "5") => (3141, 5)`
- Sum the integers digits to the decimal digits corrected by the right factor, i.e. `(3141, 5) => 3141 + 0.5 => 3141.5`

Reply:
I am using above regex to clean the string of symbols. I can achieve this by flipping the regex to filter everything that is not a digit. Let me give it a try.

Reply:
> I am using above regex to clean the string of symbols. I can achieve this by flipping the regex to filter everything that is not a digit.

This ignores my point on the risk of using strigs that we don't know beforehand to build a regular expression. If a user decides that `"["` is the prefix, the resulting regular expression is invalid, for example.

I wouldn't go down this rabbit hole.

Reply:
I am using the `[^\d]` regex to match all characters that are not digits. I am using this regex to clean the string after splitting them. But, in terms of purity while parsing `Locale.thousandSeparator` has no value anymore. Refer the second test for parse function in the new commit.

Let me know your thoughts.

Reply:
Change implemented.